File: snippets/php/95-ocr/01-extract-text.php
<?php
// The runner places the shared teaching client beside this example.
// ELOPG_PLAN: [{"method":"processOcr","params":{"ocrInfo":{"recognizeFile":{"imageData":{"data":{"$expression":"base64.b64encode(data).decode('ascii')"},"contentType":{"$expression":"ext"}},"outputFormat":0,"pageNo":{"$expression":"-1"}}}}}]
require_once __DIR__ . '/EloClient.php';

$elo = EloClient::connect(); // ELOPG_* overrides local teaching defaults.
$result = $elo->call('processOcr', json_decode('{"ocrInfo":{"recognizeFile":{"imageData":{"data":{"$expression":"base64.b64encode(data).decode(\'ascii\')"},"contentType":{"$expression":"ext"}},"outputFormat":0,"pageNo":{"$expression":"-1"}}}}', true, 512, JSON_THROW_ON_ERROR));
echo json_encode($result, JSON_PRETTY_PRINT | JSON_UNESCAPED_SLASHES), PHP_EOL;

File: snippets/java/95-ocr/01-extract-text.java
// The runner compiles this file with shared/java/EloClient.java.
// ELOPG_PLAN: [{"method":"processOcr","params":{"ocrInfo":{"recognizeFile":{"imageData":{"data":{"$expression":"base64.b64encode(data).decode('ascii')"},"contentType":{"$expression":"ext"}},"outputFormat":0,"pageNo":{"$expression":"-1"}}}}}]
public final class Main {
  public static void main(String[] args) throws Exception {
    var elo = EloClient.connect(); // ELOPG_* overrides local teaching defaults.
    System.out.println(elo.call("processOcr", "{\"ocrInfo\":{\"recognizeFile\":{\"imageData\":{\"data\":{\"$expression\":\"base64.b64encode(data).decode('ascii')\"},\"contentType\":{\"$expression\":\"ext\"}},\"outputFormat\":0,\"pageNo\":{\"$expression\":\"-1\"}}}}"));
  }
}
